### DISCUSSION TITLE
Fix Markdown scrolling off page

### DISCUSSION THREAD

**Discussion by tkg61** [open-webui/open-webui#6783] [Ideas]
# Bug Report

## Installation Method

Helm install

## Environment

- **Open WebUI Version:** 0.3.35

- **Browser (if applicable):** Chrome latest

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [X] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Markdown responses scroll off the page and should be line wrapped to fit on the page

## Actual Behavior:

Markdown scrolls off the page and is hard to read

## Description

**Bug Summary:**
Ask for a markdown response and see that it scrolls off the page: 

## Reproduction Details

**Steps to Reproduce:**

Ask for a markdown response and see the scrolling: 

See image: 

![image](https://github.com/user-attachments/assets/180f547e-c3cb-4552-9f7b-16435b05f15a)

## Additional Information

Adding this code to the browser fixes the result but up to the devs to fix how they see fit: 

```
javascript:(function(){

    let x = document.getElementsByClassName("whitespace-nowrap");

    if(x.length === 0) { return; }

    for (let i=0; i<x.length; i++) {
        x[i].classList.remove("whitespace-nowrap");
    }


})();
```

which does something like this: 

![image](https://github.com/user-attachments/assets/1e6463df-ca88-4445-ba80-5e766e03f891)

**Comment by bigRabbit01** [comment]
How do you write your back-end code, can you tell me how you implement it ?

**Comment by jeremykei** [comment]
Try change whitespace-nowrap to break-normal to effectuate css equivalent of text-wrap: auto.

File under /open-webui/src/lib/components/chat/Messages/Markdown
/MarkdownTokens.svelte:

![image](https://github.com/user-attachments/assets/c687d41d-021e-4726-b3b0-ad1aae35931f)

**Reply by Koesn** [reply]
There's no src in open-webui. Using pip installation. How to edit this?

**Comment by Koesn** [comment]
I agree, it's a real struggle to read. Fingers crossed we can go back to the normal way of text wrapping soon!

**Comment by tjbck** [comment]
Addressed in dev!
